ledger-core/v2/scripts/gen_plantuml.go

```diff
@@ -0,0 +1,562 @@
+                  unvisited = append(unvisited, ret.Args[0].Fun)
+                  uml += fmt.Sprintf("\n%s -[#DarkGreen]-> %s : %s+(StepMigration)", state.Name, ret.StepMigration, ret.Var.Fun)
+                  unvisited = append(unvisited, ret.StepMigration)
+              case "ThenRepeat":
+                  uml += fmt.Sprintf("\n%s --> %s : ThenRepeat", state.Name, state.Name)
+              case "RepeatOrJumpElse":
+                  uml += fmt.Sprintf("\n%s -[#RoyalBlue]-> %s : RepeatOr(Jump)Else", state.Name, ret.Args[2].Fun)
+                  uml += fmt.Sprintf("\n%s -[#DarkGreen]-> %s : RepeatOrJump(Else)", state.Name, ret.Args[3].Fun)
+              default:
+                  pf.diag("\n(=> (. %s %s)", ret.Var.Obj, ret.Var.Fun)
+                  for _, arg := range ret.Args {
+                      pf.diag("\n       %s", fmt.Sprintf("(. %s %s)", arg.Obj, arg.Fun))
+                  }
+                  pf.diag(")")
+              }
+
+              // -:- fn representation
+              pf.diag(fmt.Sprintf("\n(-> (. %s %s)", ret.Var.Obj, ret.Var.Fun))
+
+              for _, arg := range ret.Args {
+                  pf.diag("\n       %s", fmt.Sprintf("(. %s %s)", arg.Obj, arg.Fun))
+              }
+              pf.diag(")")
+        }
+        unvisited = SetDiff(unvisited, visited)
+    }
+
+    state_keys := make([]string, 0, len(pf.states))
+    for k := range pf.states {
+        state_keys = append(state_keys, k)
+    }
+
+    // THIS IS TEMPORARY DUPLICATE CODE FOR DEBUG PURPROSES
+    for _, state_key := range SetDiff(state_keys, visited) {
+        uml += fmt.Sprintf("\n%s : UNVISITED", state_key)
+
+        state := pf.states[state_key]
+
+        for _, ret := range state.Rets {
+            switch ret.Var.Fun {
+            case "Stop":
+                uml += fmt.Sprintf("\n%s --> [*]", state.Name)
+            case "Jump", "ThenJump":
+                uml += fmt.Sprintf("\n%s --> %s : %s", state.Name, ret.Args[0].Fun, ret.Var.Fun)
+                unvisited = append(unvisited, ret.Args[0].Fun)
+                // if nil != possible_migrations {
+                //     pf.states[ret.Args[0].Fun].Prop = append(pf.states[ret.Args[0].Fun].Prop, possible_migrations...)
+                // }
+            case "JumpExt":
+                uml += fmt.Sprintf("\n%s --> %s : %s", state.Name, ret.Args[0].Fun, ret.Var.Fun)
+                unvisited = append(unvisited, ret.Args[0].Fun)
+                uml += fmt.Sprintf("\n%s -[#DarkGreen]-> %s : %s-(StepMigration)", state.Name, ret.StepMigration, ret.Var.Fun)
+                unvisited = append(unvisited, ret.StepMigration)
+            case "ThenRepeat":
+                uml += fmt.Sprintf("\n%s --> %s : ThenRepeat", state.Name, state.Name)
+            case "RepeatOrJumpElse":
+                uml += fmt.Sprintf("\n%s -[#RoyalBlue]-> %s : RepeatOr(Jump)Else", state.Name, ret.Args[2].Fun)
```

Comment:
Move these constant strings to constant
I mean , for example, '[#RoyalBlue]-> %s : RepeatOr(Jump)Else'.
And all others. At least those what are repeated